### ISSUE TITLE
Installing with pip doesn't include addons into the package

### ISSUE THREAD

**Issue by llacroix** [OCA/OpenUpgrade#2040]
Impacted versions:
I'd guess all

Steps to reproduce:

```
git clone -b 12.0 https://github.com/OCA/OpenUpgrade.git odoo
cd odoo
pip install . 
```

Current behavior:

Starting odoo with the installed odoo it will be missing all the modules contained into odoo/addons *.py contained into odoo/odoo/addons/**/migrations aren't included also

Expected behavior:

It should be possible to install openupgrade using pip at least without having to do some manual modifications.

**Comment by llacroix** [comment]
To give a better understanding of what I'm achieving here is that I'm making openupgrade docker images and installing openupgrade as an image require to do some manual modifications to MANIFEST.in and moving root/addons to root/odoo/addons 
It's not a big change but it would certainly help building the docker image.
Also could it be possible to have tar.gz egg hosted directly from github so there wouldn'T be need to install git inside the docker image.

**Comment by pedrobaeza** [comment]
Well, installing through pip is not the main focus of this project, and there isn't / there won't be work about that specifically for OpenUpgrade. Any way, if Odoo is installable that way, OpenUpgrade basically should be, as it's an Odoo clone with certain modified processes.

I don't understand as well why do you need to do it via pip, when it's very easy to use it from source.

And we already have it working on our Docker based solution called Doodba if you don't want to waste time creating another one: https://github.com/Tecnativa/doodba. Just download the scaffolding for Docker compose and change these 2 references to point to `openupgrade`:

https://github.com/Tecnativa/doodba-scaffolding/blob/master/odoo/custom/src/repos.yaml#L13
https://github.com/Tecnativa/doodba-scaffolding/blob/master/odoo/custom/src/repos.yaml#L15

Then restore your DB inside with `docker-compose run ... pg_restore ...`, the filestore, and finally proceed with the upgrade with `docker-compose run --rm odoo -u all --stop-after-init`.

**Comment by sbidoul** [comment]
@llacroix this is not an OpenUpgrade issue, it's an Odoo issue that is tracked in https://github.com/odoo/odoo/issues/16700. 

TL;DR: `pip install -e .` should work.

**Comment by llacroix** [comment]
@sbidoul ah right 

> I don't understand as well why do you need to do it via pip, when it's very easy to use it from source.

@pedrobaeza  to speed up the build time of images eventually and reducing the dependencies required to install to build the image itself. 

I had a look at doodba but it's not a project that would really fit our team. We're trying to move away from buildout because we have too many client to manage and managing every single configuration is time consuming. The main issue is that we have many clients and when we have a change, we'd have to update all the client configurations that share the same features.
So I'm trying to come up with something that keeps the bare minimum for configurations and allow enough flexibility like installing dependencies from the module inside the addons available to the docker service.

 So most of the dependencies are delegated to the addons project instead of the configuration of the project itself. So I build 1 image to run multiple project with different addons mounted as volume instead. addons can be mounted as bind also for development and while developping, there's no need to rebuild the image once it's built. Restarting the container is just enough to update it unless you need a newer version of odoo, it could be done through "docker pull". 

A bit off topic but yeah...

ideally I'd like to be able to run jobs in gitlab like this:

```
migrations9:
   stage: v9
   script:
     - restore_db v8
     - docker run openupgrade:9.0 -d db
     - dump_d v9

migrations10:
   stage: v10
   script:
     - restore_db v9
     - docker run openupgrade:10.0 -d db
     - dump_d v10

stages:
  - v9
  - v10
```

**Comment by pedrobaeza** [comment]
@Yajo can you explain to @llacroix the Doodba design decisions and advantages of leaving out the source code and configure per project addons / repos?

**Comment by yajo** [comment]
Well, first of all this is no silver bullet; all options have their own pros and cons.

Basically if you install addons with pip, you install one addon and that includes all other addons and python dependencies in its dependency chain. However, the development and deployment environment structure and tools differ quite a bit, raising the barrier between dev and ops. Another disadvantage is that if you need several merges for a single repo, you have to do that manually. Also binary dependencies are out of the package.

If you use doodba (git and specific dependencies), you have to specify all dependencies by hand: modules, pip and apt. The advantage is that dev and ops tools and layout are mostly the same, and merging downstream is supported. I can happily say that, after some years, there hasn't been any corner case that could beat Doodba.

At the end of the day, just choose by yourself, you are lucky to have options! :wink:
